packages/bot-skeleton/src/scratch/utils/index.js

```diff
@@ -307,6 +307,7 @@ export const addDomAsBlock = (el_block, parent_block = null) => {
 export const hasAllRequiredBlocks = workspace => {
     const blocks_in_workspace = workspace.getAllBlocks();
     const trade_type_block = workspace.getAllBlocks(true).find(block => block.type === 'trade_definition_tradetype');
+    if (!trade_type_block) return false;
     const selected_trade_type = trade_type_block.getFieldValue('TRADETYPE_LIST');
```

Comment:
We have added early return in case where `trade_definition_block` is undefined, this was causing the issue at initialization of the next variable and was throwing error `cannot read properties of undefined reading: getFieldValue`